src/renderer/components/+custom-resources/init-resource-store.injectable.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import apiManagerInjectable from "../../../common/k8s-api/api-manager/manager.injectable";
+import type { CustomResourceDefinition } from "../../../common/k8s-api/endpoints";
+import { KubeApi } from "../../../common/k8s-api/kube-api";
+import { KubeObject } from "../../../common/k8s-api/kube-object";
+import { CustomResourceStore } from "./resource.store";
+
+export type InitCustomResourceStore = (crd: CustomResourceDefinition) => void;
+
+const initCustomResourceStoreInjectable = getInjectable({
+  id: "init-custom-resource-store",
+  instantiate: (di): InitCustomResourceStore => {
+    const apiManager = di.inject(apiManagerInjectable);
+
+    return (crd) => {
+      const objectConstructor = class extends KubeObject {
+        static readonly kind = crd.getResourceKind();
+        static readonly namespaced = crd.isNamespaced();
+        static readonly apiBase = crd.getResourceApiBase();
+      };
+
+      const api = apiManager.getApi(objectConstructor.apiBase)
+        ?? new KubeApi({ objectConstructor });
+
+      if (!apiManager.getStore(api)) {
+        apiManager.registerStore(new CustomResourceStore(api));
+      }
```

Comment:
All this is pretty complicated don't you think? One can leave a todo to acknowledge this transitional subquality. Or just fix it :)

Reply:
What is complicated about it?

 It is trying to ensure that all customer resource definitions for a specific cluster have a registered KubeApi instance and a registered KubeObjectStore instance. 

But it doesn't want to override the preexisting instances. Sure it doesn't currently handle when a CRD is removed but that has never been handled.

Reply:
What it does may be justified, but how it does it, not so much.

Is there not a thing you would call complicated here? Surely that cannot be :)

Reply:
I think the complication is only related to the fact that we want to auto-register. If we follow what you call HOLLYWOOD then this would probably go away I surmise.

Reply:
True! Resolved.